src/Evolution/Systems/NewtonianEuler/Sources/LaneEmdenGravitationalField.cpp

```diff
@@ -19,13 +19,18 @@ namespace Sources {
 
 void LaneEmdenGravitationalField::apply(
     const gsl::not_null<tnsr::I<DataVector, 3>*> source_momentum_density,
-    const Scalar<DataVector>& mass_density,
+    const gsl::not_null<Scalar<DataVector>*> source_energy_density,
+    const Scalar<DataVector>& mass_density_cons,
+    const tnsr::I<DataVector, 3>& momentum_density,
     const NewtonianEuler::Solutions::LaneEmdenStar& star,
-    const tnsr::I<DataVector, 3>& x) const noexcept {
+    const tnsr::I<DataVector, 3>& x) noexcept {
```

Comment:
Why is this not const? Is it because you changed the function to be static? If I understand correctly, this apply function can be static only because this class has no members, otherwise you'd need it to be non-static.

Source term classes in NewtonianEuler are used in the generic ComputeSources action in the NewtonianEuler directory, which just wraps the call for the particular source's apply member. Note that this generic action _is_ static, as required by the generic evolution code, but the source-specific apply's are not, as the classes are expected to hold members.

Reply:
You understand correctly 🙂. I am guessing that @nilsleiffischer suggested to mark this as `static` because then it can be called without instantiating a LaneEmdenGravField object.. but it does reduce consistency with the other NEuler sources. Do you have a preference here?

Reply:
The way sources are used is by creating an instance in the initial data so, unless we're violating some C++ practice, I'd revert his suggestion in order to keep consistency with the other sources.

Reply:
You can still call this function the same way as before, on an instance of the class, even if it‘s static. I didn‘t expect that to work either when I first encountered this situation, but it does :) We have been making functions static whenever possible, not sure if there‘s really a performance benefit to it though.

Reply:
@nilsleiffischer if we're not sure there's a performance benefit, I propose to undo the `static` to maintain consistency with the other sources terms. Ok with you?

Reply:
I‘m leaning the other way (I think a class shouldn’t have member functions if it has no state; making the function static doesn’t change how you can call it), but it‘s ok for me if you prefer :)

Reply:
Ok you've convinced me; I'll keep it static ;)